package/MDAnalysis/lib/log.py

```diff
@@ -296,11 +320,12 @@ def echo(self, step, **kwargs):
             return
         self.update(step, **kwargs)
         format = self.format
+        newline = not self.dynamic
         if self.step == self.numsteps:
-            if self.last_newline:
-                format = self.format[:-1] + self.last_newline
+            newline = True
         elif self.numouts % self.interval == 0:
             pass
         else:
             return
-        echo(format % vars(self))
+        echo(format.format(**vars(self)),
```

Comment:
I do not know if the progress meter was so much used so far. Though, I see 3 solutions:

* the first one is to revert back to the old % formatting, yet it is not future proof
* an other solution is to detect if the format uses the old or the new formatting, and to use format or % based on that detection (guess actually)
* add a keyword to switch back to the old formatting

Solution 2 and 3 are not completely incompatible, but mixing them may be overkill.